src/ophys_etl/modules/downsample_video/utils.py

```diff
@@ -0,0 +1,637 @@
+        quantiles: Tuple[float, float] = (0.1, 0.99),
+        reticle: bool = True,
+        speed_up_factor: int = 8,
+        tmp_dir: Optional[pathlib.Path] = None) -> None:
+    """
+    Create a video file (mp4, avi, etc.) from an HDF5 file, applying
+    downsampling and a median filter if desired.
+
+    Parameters
+    ----------
+    input_path: pathlib.Path
+        Path to the HDF5 file containing the movie data
+
+    input_hz:
+        Frame rate of the input movie in Hz
+
+    video_path: Pathlib.path
+        Path to the video file to be written
+
+    output_hz: float
+        Frame rate of the output movie in Hz (set lower than input_hz
+        if you want to apply downsampling to the movie)
+
+    kernel_size: Optional[int]
+        Size of the median filter kernel to be applied to the downsampled
+        movie (if None, no median filter will be applied)
+
+    n_processors: int
+        Number of parallel processes to be used when processing the movie
+
+    quality: int
+        A value between 0-9 (inclusive) denoting the quality of the movie
+        to be written (higher number means better quality)
+
+    quantiles: Tuple[float, float]
+        The quantiles to which to clip the movie before writing it to video
+
+    reticle: bool
+        If True, add a grid of red lines to the movie to guide the eye
+
+    speed_up_factor: int
+        Factor by which to speed up the movie *after downsampling* when writing
+        to video (in case you want a smaller file that can be played back
+        faster)
+
+    tmp_dir: Optional[pathlib.Path]
+        Scratch directory to use during processing. When applying the median
+        filter, the code writes the filtered movie to disk, rather than try
+        to keep two copies of the movie in memory. This gives the user the
+        option to specify where the scratch copy of the movie is written.
+        If None, the scratch movie will be written to the system's default
+        scratch space.
+
+    Returns
+    -------
+    None
+        Output is written to the specified movie file
+    """
+
+    with tempfile.TemporaryDirectory(dir=tmp_dir) as this_tmp_dir:
+        tmp_h5 = tempfile.mkstemp(dir=this_tmp_dir, suffix='.h5')[1]
+        tmp_h5 = pathlib.Path(tmp_h5)
+        print(f'writing h5py to {tmp_h5}')
+
+        create_downsampled_video_h5(
+            input_path, input_hz,
+            tmp_h5, output_hz,
+            kernel_size,
+            n_processors)
+
+        print(f'wrote temp h5py to {tmp_h5}')
+
+        (min_val,
+         max_val) = _min_max_from_h5(tmp_h5, quantiles)
+
+        video_array = _video_array_from_h5(
+                tmp_h5,
+                min_val,
+                max_val,
+                reticle)
+
+        tmp_h5.unlink()
+
+        _write_array_to_video(
+            video_path,
+            video_array,
+            int(speed_up_factor*output_hz),
+            quality)
+
+
+def create_side_by_side_video(
```

Comment:
I believe this would be better off as a method of the `SideBySideDownsampler`. It is specific to side by side video, whereas this module should contain helper functions for video in general